packages/formstr-app/src/containers/ResponsesNew/components/ResponseDetailModal.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useState, useEffect } from 'react';
+import { Modal, Descriptions, Typography, Button, Space } from 'antd';
+import { Event, nip19, nip44 } from 'nostr-tools';
+import { Tag, Field } from '../../../nostr/types'; 
+
+const { Text } = Typography;
+
+type ResponseDetailItem = {
+  key: string; 
+  question: string;
+  answer: string;
+};
+interface ResponseDetailModalProps {
+  isVisible: boolean;
+  onClose: () => void;
+  event: Event | null; 
+  formSpec: Tag[] | null | undefined; 
+  editKey: string | undefined | null; 
+}
+export const ResponseDetailModal: React.FC<ResponseDetailModalProps> = ({
+  isVisible,
+  onClose,
+  event,
+  formSpec,
+  editKey,
+}) => {
+  const [processedData, setProcessedData] = useState<ResponseDetailItem[]>([]);
+  const [metaData, setMetaData] = useState<{ author?: string, timestamp?: string }>({});
+
+  useEffect(() => {
+    if (isVisible && event) {
+      const authorNpub = nip19.npubEncode(event.pubkey);
+      const timestamp = new Date(event.created_at * 1000).toLocaleString();
+      setMetaData({ author: authorNpub, timestamp });
+      setProcessedData([{ key: 'loading', question: 'Processing...', answer: '...' }]); 
+    } else {
+      setProcessedData([]);
+      setMetaData({});
+    }
+  }, [isVisible, event, formSpec, editKey]); 
+  const getInputs = (responseEvent: Event): Tag[] => {
```

Comment:
should we move it to a common place cause i believe we use it in the responses table as well